src/recaptcha-verification/recaptcha-verification.service.ts

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import axios from 'axios';
+import { ApiConfigService } from '../api-config/api-config.service';
+
+@Injectable()
+export class RecaptchaVerificationService {
+  constructor(private readonly configService: ApiConfigService) {}
+
+  async verify(recaptcha: string): Promise<boolean> {
+    const recaptchaSecret = this.configService.get<string>(
+      'RECAPTCHA_SECRET_KEY',
+    );
+
+    const recaptchaBaseUrl = 'https://www.google.com/recaptcha/api/siteverify';
+    const recaptchaVerificationUrl = `${recaptchaBaseUrl}?secret=${recaptchaSecret}&response=${recaptcha}`;
+
+    try {
+      const isVerified = await axios
+        .post<GoogleRecaptchaVerificationJsonResponse>(recaptchaVerificationUrl)
+        .then(({ data }) => {
```

Comment:
Is this `.then` needed? Or could we just return `response.data.success`?

Reply:
`response.data.success` works - updated